scripts/functions/installer

```diff
@@ -1019,150 +1021,175 @@ setup_user_profile_check()
-    done
-    found_rc=()
-  fi
-  if
-    (( rvm_auto_dotfiles_flag == 1 || ${#found_login[@]} == 0 ))
-  then
-    rvm_out "    Adding rvm loading line to ${target_login[*]}."
-    for profile_file in "${target_login[@]}"
-    do
-      [[ -f "$profile_file" ]] ||
-      {
-        touch "$profile_file"
-        if
-          [[ "$profile_file" == "$HOME/.bash_"* && -f "$HOME/.profile" ]]
-        then
-          printf "%b" "
-[[ -s \"\$HOME/.profile\" ]] && source \"\$HOME/.profile\" # Load the default .profile
-" >> "$profile_file"
-        fi
-      }
-      printf "%b" "
-[[ -s \"$local_rvm_path/scripts/rvm\" ]] && source \"$local_rvm_path/scripts/rvm\" # Load RVM into a shell session *as a function*
-" >> "$profile_file"
-    done
-  else
-    setup_user_profile_summary found_login "sourcing"
-  fi
+  rvm_out "    Loading RVM in ${profile_login[*]}"
+
+  for file in "${profile_login[@]}"
+  do
+    # Remove old style entries (added by RVM up to 1.29.2)
+    if
+      [[ -f "$file" ]]
+    then
+      if
+        __rvm_grep scripts/rvm "$file" >/dev/null
+      then
+        rvm_debug "Removing RVM loading line from ${file}"
+        __rvm_sed_i "${file}" -e '/source.*scripts\/rvm/ d; /\. .*scripts\/rvm/ d;'
+        setup_user_profile_delete_duplicate_blank_lines "$file"
+      fi
+    fi
+
+    # Add new entries
+    [[ -f "$file" ]] ||
+    {
+      touch "$file"
+      if
+        [[ "$file" == "$HOME/.bash_"* && -f "$HOME/.profile" ]]
+      then
+        printf "%b" "
+# Load the default .profile
+[[ -s \"\$HOME/.profile\" ]] && source \"\$HOME/.profile\"
+" >> "$file"
+      fi
+    }
+
+    printf "%b" "
+# Load RVM into a shell session *as a function*
+[[ -s \"$local_rvm_path/scripts/rvm\" ]] && source \"$local_rvm_path/scripts/rvm\"
+" >> "$file"
+  done
+}
+
+setup_user_profile_verify()
+{
+  for file in "${profile_rc[@]}"
+  do
+    if
+      ! __rvm_grep PATH=.*$local_rvm_path/bin "$file" >/dev/null
+    then
+      rvm_warn "    RVM PATH line not found in ${file}, rerun this command with '--auto-dotfiles' flag to fix it."
+    fi
+  done
+
+  for file in "${profile_login[@]}"
+  do
+    if
+      ! __rvm_grep scripts/rvm "$file" >/dev/null
+    then
+      rvm_warn "    RVM sourcing line not found in ${file}, rerun this command with '--auto-dotfiles' flag to fix it."
+    fi
+  done
 }
 
 setup_user_profile()
 {
   setup_user_profile_check || return 0
 
   export user_profile_file
-  \typeset -a search_list search_list_mksh search_list_bash search_list_zsh
-  \typeset -a target_rc target_login found_rc found_login
+  \typeset -a profile_all profile_rc profile_login
   \typeset etc_profile_file profile_file local_rvm_path local_rvm_path_sed
   local_rvm_path="${rvm_path/#$HOME/\$HOME}"
   local_rvm_path_sed="\\${local_rvm_path//\./\\.}"
   local_rvm_path_sed="${local_rvm_path_sed//\//\/}"
 
-  setup_user_profile_detect
-  setup_user_profile_rc
-  setup_user_profile_login
+  if
+    (( rvm_auto_dotfiles_flag == 1 ))
```

Comment:
if we keep updating the line at every install it might mess up things? what if user adds something that depends on RVM and we move our code after his addition?